lib/register.php

```diff
@@ -82,39 +138,146 @@ function gutenberg_add_edit_links_filters() {
  *
  * @since 0.1.0
  *
- * @param  array $actions Post actions.
- * @param  array $post    Edited post.
+ * @param  array   $actions Post actions.
+ * @param  WP_Post $post    Edited post.
  *
  * @return array          Updated post actions.
  */
 function gutenberg_add_edit_links( $actions, $post ) {
-	$can_edit_post = current_user_can( 'edit_post', $post->ID );
-	$title = _draft_or_post_title( $post->ID );
-	$post_type = get_post_type( $post );
+	if ( ! gutenberg_can_edit_post( $post->ID ) ||
+			'trash' === $post->post_status ||
+			! post_type_supports( $post->post_type, 'editor' ) ||
+			! apply_filters( 'gutenberg_add_edit_link_for_post_type', true, $post->post_type, $post ) ) {
+		return $actions;
+	}
 
-	if ( $can_edit_post && 'trash' !== $post->post_status && apply_filters( 'gutenberg_add_edit_link_for_post_type', true, $post_type, $post ) ) {
-		// Build the Gutenberg edit action. See also: WP_Posts_List_Table::handle_row_actions().
-		$gutenberg_url = menu_page_url( 'gutenberg', false );
-		$gutenberg_action = sprintf(
+	if ( gutenberg_post_has_blocks( $post->ID ) ) {
+		remove_filter( 'get_edit_post_link', 'gutenberg_filter_edit_post_link', 10 );
+		add_filter( 'get_edit_post_link', 'gutenberg_filter_edit_post_link', 10, 3 );
```

Comment:
As it stands right now, this `if` condition with add/remove of the `get_edit_post_link` filter is just removing and adding something right back, so it has no purpose right?